src/tr/merlinscans/src/eu/kanade/tachiyomi/extension/tr/merlinscans/MerlinScans.kt

```diff
@@ -1,12 +1,457 @@
+            val seriesLink = element.select("h3.series-title a").attr("href")
+            setUrlWithoutDomain(normalizeUrl(seriesLink))
+            title = element.select("h3.series-title a").text()
+            thumbnail_url = element.select("div.series-cover-thumbnail img").attr("src")
+
+            val latestChapter = element.select("div.chapter-item").first()
+            if (latestChapter != null) {
+                val chapterNumber = latestChapter.select("span.chapter-number").text()
+                val chapterTime = latestChapter.select("span.chapter-time").text()
+                description = "Son bölüm: $chapterNumber ($chapterTime)"
+            }
+        }
+    }
+
+    override fun latestUpdatesNextPageSelector(): String? = null
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.isNotEmpty()) {
+            val url = "$baseUrl/ajax/search-preview.php".toHttpUrl().newBuilder()
+                .addQueryParameter("q", query)
+                .build()
+            return GET(url, headers)
+        }
+
+        val url = "$baseUrl/all-series.php".toHttpUrl().newBuilder()
+            .addQueryParameter("page", page.toString())
+
+        var sortApplied = false
+
+        filters.forEach { filter ->
+            when (filter) {
+                is TypeFilter -> {
+                    if (filter.state != 0) {
+                        url.addQueryParameter("type", filter.toUriPart())
+                    }
+                }
+
+                is CategoryFilter -> {
+                    if (filter.state != 0) {
+                        url.addQueryParameter("category", filter.toUriPart())
+                    }
+                }
+
+                is StatusFilter -> {
+                    if (filter.state != 0) {
+                        url.addQueryParameter("status", filter.toUriPart())
+                    }
+                }
+
+                is SortFilter -> {
+                    if (filter.state != 0) {
+                        url.addQueryParameter("sort", filter.toUriPart())
+                        sortApplied = true
+                    }
+                }
+
+                else -> {}
+            }
+        }
+
+        if (!sortApplied) {
+            url.addQueryParameter("sort", "views")
+        }
+
+        return GET(url.build(), headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val requestUrl = response.request.url.toString()
+
+        if (requestUrl.contains("search-preview.php")) {
+            val body = response.body.string()
+            return parseSearchJson(body)
+        } else {
+            val document = response.asJsoup()
+            return parseSearchPage(document, response)
+        }
+    }
+
+    private fun parseSearchJson(json: String): MangasPage {
+        val mangas = mutableListOf<SManga>()
+
+        try {
+            val jsonObject = org.json.JSONObject(json)
```

Comment:
Use `kotlinx.serialization` with DTO instead of `org.json`, also remove the try-catch